### PR TITLE
Fixed focus issues.

### LINKED ISSUE #203: PhotoCapture: focus issues
https://developer.microsoft.com/en-us/microsoft-edge/testdrive/demos/photocapture/

1.Looks like one has to click on the video to take a photo. It is not possible to reach this via keyboard
2. when switching camera with the switch button, the focus is lost after pressing it, so one has to tab back to it to switch it again (looks like tab focus moves to the first tab stop - the browser chrome)

### PR DESCRIPTION
**Pull request by sarvaje**
Fixed #203

### CODE REVIEW

File: photocapture/scripts/demo.js [hunk 1]
```diff
@@ -64,23 +64,43 @@
...
             }
             canvas.removeEventListener('click', savePhoto);
+            canvas.removeEventListener('keydown', savePhotoWithKey, false);
             document.getElementById('photoViewText').innerHTML = '';
             photoReady = false;
         }
     };
 
+    var isEnterKey = function (evt) {
+        var charCode = (typeof evt.which === "number") ? evt.which : evt.keyCode;
```
Comment (alrra) [inline photocapture/scripts/demo.js]: Use `'number'` in order to be consistent with the rest of the code.

File: photocapture/scripts/demo.js [hunk 2]
```diff
@@ -93,25 +113,36 @@
...
+        switchButton.disabled = true;
+        if (currentCam !== null) {
             currentCam++;
-            if(currentCam >= webcamList.length) {
+            if (currentCam >= webcamList.length) {
                 currentCam = 0;
             }
             var video = document.getElementById('videoTag');
-            if (typeof(video.srcObject) !== 'undefined') video.srcObject = null;
+            if (typeof (video.srcObject) !== 'undefined') video.srcObject = null;
```
Comment (alrra) [inline photocapture/scripts/demo.js]: Drop parentheses surrounding `video.srcObject`.

File: photocapture/scripts/demo.js [hunk 3]
```diff
@@ -93,25 +113,36 @@
...
     // nextWebCam() - Function to rotate through the webcam device list
     // 1. Release the current webcam (if there is one in use)
     // 2. Call getUserMedia() to access the next webcam
 
-    var nextWebCam = function() {
-        document.getElementById('switch').disabled = true;
-        if(currentCam !== null) {
+    var nextWebCam = function () {
+        var switchButton = document.getElementById('switch');
+        switchButton.disabled = true;
```
Comment (alrra) [inline photocapture/scripts/demo.js:137]: `autocomplete="off"` will need to be added to the `switch` button as Firefox, after reloading the page, will keep a button that has been disabled with JavaScript, disabled (see: https://bugzilla.mozilla.org/show_bug.cgi?id=654072).

File: photocapture/scripts/demo.js [hunk 4]
```diff
@@ -64,23 +64,43 @@
...
+            canvas.removeEventListener('keydown', savePhotoWithKey, false);
             document.getElementById('photoViewText').innerHTML = '';
             photoReady = false;
         }
     };
 
+    var isEnterKey = function (evt) {
+        var charCode = (typeof evt.which === "number") ? evt.which : evt.keyCode;
+
+        if (charCode !== 13) {
```
Comment (alrra) [inline photocapture/scripts/demo.js]: Shouldn't pressing the spacebar (32) also work?

File: photocapture/scripts/demo.js [hunk 5]
```diff
@@ -127,12 +158,15 @@
                 height: 360,
                 deviceId: { exact: webcamList[currentCam] }
             }
-        }).then(initializeVideoStream).catch(getUserMediaError);
+        }).then(function () {
+            initializeVideoStream();
```
Comment (alrra) [inline photocapture/scripts/demo.js]: You are not sending the stream to the `initializeVideoStream` function anymore. 

This should be something such as:

```js
.then(function (stream) {
     initializeVideoStream(stream);

```

File: photocapture/scripts/demo.js (hunk 4 above)
Comment (sarvaje) [reply photocapture/scripts/demo.js]: ok!

**Comment by sarvaje** [comment]
@alrra Done!

**Comment by melanierichards** [comment]
Woo hoo, works for me!

**Pull request merged**